packages/manager/src/features/Account/utils.ts

```diff
@@ -41,16 +62,33 @@ export const getAccessRestrictedText = (userType: UserType | null) => {
   } to request the necessary permission.`;
 };
 
+/**
+ * Determine whether the user has restricted access to a specific resource.
+ *
+ * @example
+ * // If account access does not equal 'read_write', the user has restricted access.
+ * isRestrictedGlobalGrantType({
+ *   globalGrantType: 'account_access',
+ *   allowableGrantLevel: 'read_write',
+ *   grants,
+ *   profile
+ * });
+ * // Returns: true
+ */
 export const isRestrictedGlobalGrantType = ({
   globalGrantType,
   grants,
+  permittedGrantLevel,
   profile,
-}: {
-  globalGrantType: GlobalGrantTypes;
-  grants: Grants | undefined;
-  profile: Profile | undefined;
-}): boolean => {
-  return Boolean(profile?.restricted) && !grants?.global[globalGrantType];
+}: IsRestrictedGlobalGrantType): boolean => {
```

Comment:
```suggestion
}: RestrictedGlobalGrantType): boolean => {
```
Related to suggestion above.